### DISCUSSION TITLE
What’s a practical way to process (and bundle?) inline styles?

### DISCUSSION THREAD

**Discussion by ed-ponce** [11ty/eleventy#2866] [Q&A]
###### Context

I’m trying to move away from SASS into vanilla (and future CSS). I tried [eleventy-plugin-lightningcss](https://github.com/5t3ph/eleventy-plugin-lightningcss) and I’m really enjoying it (thanks @5t3ph).

Though, I’m looking for a way to practically write and process styles inline to maintain scoped styles within its corresponding markup (be it template, layout or include). I want to emulate a similar developer experience to that of [Astro](https://astro.build) and the way it bundles and scopes styles, inline and otherwise.

###### Possible solutions

There’s Vite, which would handle all of my needs. I could integrate [Slinkity](https://slinkity.dev), but I’m not interested in islands or components right now. And there’s [eleventy-plugin-vite](https://github.com/11ty/eleventy-plugin-vite), the official integration.

There’s Parcel, which would process my CSS with Lightning CSS (which seems much better than Vite’s PostCSS integration). But I don’t know how I would integrate it with Eleventy.

And then there are shortcodes. [eleventy-plugin-bundle](https://github.com/11ty/eleventy-plugin-bundle) seems like the better way to implement this approach; highly configurable, DIY, pure Eleventy-style. But I also found the unmaintained [eleventy-plugin-helmet](https://github.com/vseventer/eleventy-plugin-helmet).

---

> Where would you start?  
> I’ve never worked with any of these cited technologies before.

**Comment by bobmonsour** [comment]
It's a great question and a great goal. I recently wrote a blog post about my path to inlining CSS. The site is very simple and I'm not doing any scoping of styles within templates. My goal was to move away from Sass and to minify and inline my small bit of css. Here's a link to the post:

[https://www.bobmonsour.com/posts/the-evolution-of-my-CSS-pipeline-in-eleventy/](https://www.bobmonsour.com/posts/the-evolution-of-my-CSS-pipeline-in-eleventy/)

I think that the [eleventy-plugin-bundle](https://github.com/11ty/eleventy-plugin-bundle) is very attractive and I think it will be my next step in the evolution, allowing me to place CSS closer to the templates that it's laying out. 

That said, I don't t think you get any sort of scoping capability with the bundle other than what you bring yourself with class names. 

To get "real" scoped CSS in eleventy, I think you'd have to go the [WebC](https://www.11ty.dev/docs/languages/webc/) route. See: [webc:scoped](https://www.11ty.dev/docs/languages/webc/#webcscoped). 

I don't currently have a need (yet) to start using WebC, but I've read the docs and it's very interesting.

I hope that's helpful.
